### ISSUE TITLE
SPENDMULTISIG error on bitcoin testnet

### ISSUE THREAD

**Issue by wliyongfeng** [trezor/connect#410]
Following code is my function to sign `P2SH` transaction on testnet:

```javascript

async function multisig() {
  const outputs = [
    {
      address: '2MtZtB1fa2y92eav2eT3U9gawaT1nZfH6PZ',
      amount: '990000',
      script_type: 'PAYTOADDRESS',
    }
  ]

  const multisig = {
    pubkeys: [
      {
        node: "tpubD6NzVbkrYhZ4WY96EeXWotzWe8yUuqaYmvbGeJCLzYrbqFXCHb8TfUUJ5TMANqMdnn5bYr1YghnF9so4vGgHveT69VSYoN2uaCeC8cRDM9H",
        address_n: [(44 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      },
      {
        node: "tpubD6NzVbkrYhZ4WY96EeXWotzWe8yUuqaYmvbGeJCLzYrbqFXCHb8TfUUJ5USfBwRwWqGD7xBXauiNzxj3HSZvxJYyZxqnfoEsLhvKxuSYnJu",
        address_n: [(44 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      },
      {
        node: "tpubDHeS8MCAkLvohjQJt1hKDHwBtPvgoqbhhKSKK5t4QSR99hz8rctYu2mDdoZLHSVr5X3ChMacuU8xzKZBYXMg27PFyLCLGpHHom92s7NYmHK",
        address_n: [(44 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      }
    ],
    signatures: ["", "", ""],
    m: 2
  }

  const inputs = [
    {
      address_n: [(48 | 0x80000000) >>> 0, 0, 0],
      prev_index: 1,
      prev_hash: 'c5e170d380d062234692d1b35f7095666be55a6fc546b27fd9bde3f360c72f8a',
      script_type: 'SPENDMULTISIG',
      amount: '1000000',
      multisig
    }
  ]

  const param = {
    coin: 'TEST',
    inputs,
    outputs,
  }

  const result = await TrezorConnect.signTransaction(param)

  console.log(result)
}

```

and `tpubDHeS8MCAkLvohjQJt1hKDHwBtPvgoqbhhKSKK5t4QSR99hz8rctYu2mDdoZLHSVr5X3ChMacuU8xzKZBYXMg27PFyLCLGpHHom92s7NYmHK` is the `xpub` from my trezor.

But the function always return error

```javascript
{
"payload": {
"error":"Failed to compile input",
"code":"Failure_ProcessError"
},
"id":1,
"success":false
}
``` 

What I have tried to fix:

1.  change `address_n` to `[(48 | 0x80000000) >>> 0]`, `[(44 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0]`
2. change signatures of multisig to `["", ""]`

But all gives the same error.

I guess `address_n` in the input is wrong, but what should be the right one for my code(in bitcoin testnet)?

**Comment by wliyongfeng** [comment]
BTW, I use [bitcoinjs-lib](https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/test/integration/addresses.js#L39) to generate the `P2SH` address.

**Comment by szymonlesisz** [comment]
your input is pointing to Bitcoin instead of Testnet. Read more about bip44:
https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki

`address_n: [(48 | 0x80000000) >>> 0, 0, 0]`

should be:

`address_n: [(48 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0]`

**Comment by wliyongfeng** [comment]
> your input is pointing to Bitcoin instead of Testnet. Read more about bip44:
> https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
> 
> `address_n: [(48 | 0x80000000) >>> 0, 0, 0]`
> 
> should be:
> 
> `address_n: [(48 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0]`

@szymonlesisz Thanks for quick response. I think you are right with the `address_n`.

I changed the `address_n` to `[(48 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0]` as your suggestion. But unfortunately, it still fails with the same error.

My new code:

```javascript
async function multisig() {
  const outputs = [
    {
      address: 'mm6UrTsf1pmxid465CGNQ6JiwuF11uvYUZ',
      amount: '10000',
      script_type: 'PAYTOADDRESS',
    }
  ]

  const multisig = {
    pubkeys: [
      {
        node: "tpubD6NzVbkrYhZ4WY96EeXWotzWe8yUuqaYmvbGeJCLzYrbqFXCHb8TfUUJ5TMANqMdnn5bYr1YghnF9so4vGgHveT69VSYoN2uaCeC8cRDM9H",
        address_n: [],
      },
      {
        node: "tpubD6NzVbkrYhZ4WY96EeXWotzWe8yUuqaYmvbGeJCLzYrbqFXCHb8TfUUJ5USfBwRwWqGD7xBXauiNzxj3HSZvxJYyZxqnfoEsLhvKxuSYnJu",
        address_n: [],
      },
      {
        node: "tpubDHeS8MCAkLvohjQJt1hKDHwBtPvgoqbhhKSKK5t4QSR99hz8rctYu2mDdoZLHSVr5X3ChMacuU8xzKZBYXMg27PFyLCLGpHHom92s7NYmHK",
        address_n: [(44 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      }
    ],
    signatures: ["", "", ""],
    m: 2
  }

  const inputs = [
    {
      address_n: [(48 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      prev_index: 1,
      prev_hash: 'c5e170d380d062234692d1b35f7095666be55a6fc546b27fd9bde3f360c72f8a',
      script_type: 'SPENDMULTISIG',
      multisig
    },
    {
      address_n: [(48 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      prev_index: 1,
      prev_hash: '7d58c6b7c049ce0fdb9f06d45c3d4fee3242d68a825bffc4322877d0c1bf2583',
      script_type: 'SPENDMULTISIG',
      multisig
    },
    {
      address_n: [(48 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      prev_index: 0,
      prev_hash: '6faa9749afa13f78111503c5009ac7e0e36976f22286174e6dc80b893c44ea48',
      script_type: 'SPENDMULTISIG',
      multisig
    }
  ]

  const param = {
    coin: 'TEST',
    inputs,
    outputs,
  }

  const result = await TrezorConnect.signTransaction(param)

  console.log(result)
}
```

My trezor popup with 

1.
```
Confirm sending 0.0001 TEST to mm6Ur..
```

2.
```
Really send 0.03 from your wallet? 
Fee included: 0.0299 TEST
```

Could you find any problem with my new code?
Any problem with the `tpub...` in the multisig?

Append: I changed the tpub to the format described [here](https://github.com/trezor/connect/issues/362#issuecomment-475188482). So the multisig is:

```javascript
  const multisig = {
    pubkeys: [
      {
        node: "tpubD6NzVbkrYhZ4WLczPJWReQycCJdd6YVWXubbVUFnJ5KgU5MDQrD998ZJLTqe6qgq2badiDXjqBwF7e6oaWB6vXuE6ZtU2TGKgYQKdSxY4bb",
        address_n: [],
      },
      {
        node: "tpubD6NzVbkrYhZ4WLczPJWReQycCJdd6YVWXubbVUFnJ5KgU5MDQrD998ZJLUw8uwm8kemFHKhijPsNxj2mwg4jxC17X3HhttUHT3gTTgzTmEk",
        address_n: [],
      },
      {
        node: "tpubDHeS8MCAkLvohjQJt1hKDHwBtPvgoqbhhKSKK5t4QSR99hz8rctYu2mDdoZLHSVr5X3ChMacuU8xzKZBYXMg27PFyLCLGpHHom92s7NYmHK",
        address_n: [(44 | 0x80000000) >>> 0, (1 | 0x80000000) >>> 0, (0 | 0x80000000) >>> 0, 0, 0],
      }
    ],
    signatures: ["", "", ""],
    m: 2
  }
```
still produce the same error.

**Comment by wliyongfeng** [comment]
It's the problem of `address_n`. Solved now.
